src/cloud-api-adaptor/pkg/securecomms/ppssh/ppssh.go

```diff
@@ -0,0 +1,304 @@
+package ppssh
+
+import (
+	"bytes"
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"errors"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"sync"
+
+	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/securecomms/sshproxy"
+	"github.com/confidential-containers/cloud-api-adaptor/src/cloud-api-adaptor/pkg/securecomms/sshutil"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	PROVEN_PP_PRIVATE_KEY_PATH  = "/tmp/provenPpPrivateKey"
+	PROVEN_WN_PUBLIC_KEY_PATH   = "/tmp/provenTePublicKey"
+	UNPROVEN_WN_PUBLIC_KEY_PATH = "/tmp/unprovenTePublicKey"
+	SIGNELTON_PATH              = "/tmp/sshSingleton"
+	PP_SID                      = "pp-sid/"
+	PP_PRIVATE_KEY              = PP_SID + "privateKey"
+	WN_PUBLIC_KEY               = "sshclient/publicKey"
+)
+
+func k8sPhase(listener net.Listener, inbounds sshproxy.Inbounds, outbounds sshproxy.Outbounds, kubernetesPhaseConfig *ssh.ServerConfig) {
+	log.Printf("Kubernetes Phase: waiting for client to connect\n")
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	var peer *sshproxy.SshPeer
+	for peer == nil {
+		nConn, err := listener.Accept()
+		if err != nil {
+			log.Fatal("failed to accept incoming connection (Kubernetes Phase): ", err)
+		}
+		//ctx, cancel := context.WithCancel(context.Background())
+		//defer cancel()
+
+		log.Printf("Kubernetes client connected\n")
+		peer, err = KubernetesSShService(ctx, nConn, kubernetesPhaseConfig)
+		if err != nil {
+			log.Printf("Failed k8sPhase: %s", err)
```

Comment:
Other messages use the wording `Kubernetes Phase`, not `k8sPhase`.

Why is this not a `Fatal`? I think it's because you are retrying. If so:

```suggestion
			log.Printf("Retrying after Kubernetes phase failed with: %s", err)
```

But then I do not understand when the loop ends. Does it keep retrying forever?

Reply:
This is Peer Pod code. 
The Peer Pod (PP) has already obtained keys during attestation phase and is now at Kubernetes phase. 
Once in this phase, the PP has nothing else to do apart from waiting forever for a Worker Node (WN)  with the right credentials to connect to it via an ssh channel. It retries forever. 
Panic with Fatal is not useful. Graceful termination the forwarder is not useful as well. The only thing to wait for a connection from a real WN that will be successful (there may be intermediate attempts from hacker that do not have the right credentials to connect which will cause failures). Also the WN may disconnect due to network of WN side issues. It may later reconnect with the same credentials.

I will fix the log message as suggested.